### DISCUSSION TITLE
Maple Island: Comprehensive List

### DISCUSSION THREAD

**Discussion by LuneArk** [sewil/OpenMG#84] [Quests]
There were some interesting things that I have noticed along the way, but that doesn't necessarily mean they are incorrect and require fixing. If anything, more research into them would be a good idea. I'm just a bit skeptical, since my memory is probably just as hazy as everyone else's. It's been almost two decades after all.

**Quests:**
- NPC dialogue needs work. There are many spelling/grammatical errors for both casual and quest-related dialogue. Some words are also partly cut off with its second half moved to the next line. Are the dialogues machine translated? If so, what is the best/closest approach to authenticity that should be taken if there are no reliable sources that documented the true text?

- The quest "Sarah's Mirror" implies that Sarah should be one of the NPCs in the quest, but the NPC is named Sera instead. According to [this](http://wayback.hidden-street.net/quests/mpislandquest_001.php), the error _should_ be vanilla. Need to do more research? An early part of the quest had Heena's second dialogue window say to press the Space Bar to jump. However, the jump key is set on the Alt key by default on all new characters. Even the first dialgoue window from the NPC Maple Administrator says jump is on the Alt key. How should we proceed?

- The quest "Roger's Apple" isn't labeled as a quest. Regardless, the quest has Roger taking 50% of your HP. Roger hands you a Roger's Apple, which heals 30 HP. This wouldn't be seen as an issue, since all new characters start with 50 HP, and with the apple healing more than the HP that was taken away. The map Roger is in, including the map before, only allows for one-way travel. It's impossible for characters to have gained any levels during this timeframe that could increase their HP to the point where having 50% of it being taken would be too much for the apple to replenish.

- [x] The quest "Please Bring This Letter To Lucas" gives you a hat as a reward. However, it gives absolutely no stats.

- [x] #69

**Some other things I've noticed ranging from high to low priority:**
- To reiterate, NPC dialogue needs work, some more than others, like NPC Robin.

- Forfeiting a quest does not take the quest item away from your inventory. Deciding to accept the quest again could mean that NPCs might give you the same quest item to be used. The quest items are stackable, but unsure if they all have this property. When this quest item is consumed as a part of completing the quest, only one of it will be taken. You will still have the other in your inventory.

- [x] #39

- Speaking of Jr. Sentinel Shellpieces, I notice that it appeared slightly below the ground as a monster drop. I believe I was still able to pick it up? I can't remember. I cannot do any more tests right now because all the Jr. Sentinels are currently extinct on the map due to [this](https://github.com/sewil/OpenMG/issues/30) issue. However, I do remember that it was re-entering the map that I could see a large pile of Jr. Sentinel Shellpieces littered across the ground, since I wouldn't normally be able to see them after having already completed the quest. Otherwise, starting the quest and killing for the quest item would appear to drop correctly above the ground, I believe.

- [ ] NPC Nina's nameplate in Maple Road: Mushroom Town appears misaligned.

- [x] #81

- Some benches in monster maps can be sat on, which was definitely a thing I remember back then. However, the bench on the far right of the Rainbow Street: The Field East of Amherst map cannot be sat on, which I remember it could. However, I can sit on the bench on Maple Road: East Entrance to Mushroom Town. Maybe I was playing a slightly newer version of game at the time? Need to do research.

- [x] #80

**Comment by LuneArk** [comment]
Here is an example of the Jr. Sentinel Shellpiece.

![Jr  Sentinel Shellpiece](https://github.com/user-attachments/assets/f8beeef1-3fd6-4a23-81e3-b2dac0057fef)

**Comment by sewil** [comment]
>NPC dialogue needs work. There are many spelling/grammatical errors for both casual and quest-related dialogue. Some words are also partly cut off with its second half moved to the next line. Are the dialogues machine translated? If so, what is the best/closest approach to authenticity that should be taken if there are no reliable sources that documented the true text?

The quest dialogue is vanilla. Only the scripts are translated.

>The quest "Sarah's Mirror" implies that Sarah should be one of the NPCs in the quest, but the NPC is named Sera instead.

Yes, all quest dialogue and quest names are vanilla as it is set in the Data.wz.

>The quest "Roger's Apple" isn't labeled as a quest.

That's right, Roger's Apple is a script quest, meaning it is not set in the Data.wz but server-side. This is probably due to the quest halving your HP, which can't be done using pure quest functionality.

>Forfeiting a quest does not take the quest item away from your inventory.

This is vanilla, I've never encountered a quest taking away your quest items when you forfeit it. The best it'll do is take them away when you accept it again, but that depends on the quest iirc.

>Speaking of Jr. Sentinel Shellpieces, I notice that it appeared slightly below the ground as a monster drop.

This is also vanilla, that's how it was displayed in old maple. It's a bit quirky but adds to the authenticity I guess.

>Some benches in monster maps can be sat on, which was definitely a thing I remember back then

Pretty sure those benches are just cosmetic. You can walk on them and they're pretty big so I think it might have been a higher version in that case.
